lib/core/src/control/callback.rs

```diff
@@ -66,27 +57,32 @@ impl Guard {
 /// handle lifetimes are strictly connected. As soon a handle is dropped, the callback is removed
 /// as well.
 #[derive(Derivative)]
-#[derivative(Debug, Default)]
-pub struct CallbackRegistry {
+#[derivative(Debug)]
+pub struct CallbackRegistry<F> {
     #[derivative(Debug="ignore")]
-    callback_list: Vec<(Guard, Box<dyn FnMut()>)>
+    callback_list: Vec<(Guard, F)>
```

Comment:
Something weird is done here. The callback registry is no longer a callback registry, because it does not collect callbacks anymore, but some arbitrary type F. And `run_all` is in fact a function _running one callback for all Fs_, not _run all callbacks_

Reply:
Ok, now I see why you did it that way, but I think it's much better to add argument type as a parameter. If someone will need to pass more argument, he can use tuple or create struct with named parameters.

Reply:
I agree, this is no longer callback registry. As I've written earlier, if you have problem making it generic over the callback types, please make copy of the original implementation, name it `CallbackRegistry1`, which will be parametrized with 1 parameter (the type of argument to every callback) and I will merge these implementations later. But we should not change the API of the original callback registry, as the API does not behave right now like a callback firing utility.

Reply:
This is the way I found to be able to send multiple arguments to our callbacks. Unfortunately it removes some constraints and makes it possible to store any kind of type in the registry.

`#![fn_traits]` looks promising for our APIs, but I found in my tests it's unstable.

I will leave `CallbackRegistry` with the feel and looks of a callback firing utility and I will provide create `CallbackRegistry1<T:Copy>`. Please note that `T` has a `Copy` constraint, because when firing it with `run_all`, it tries to move the value and, well, I need a copy of the same argument for every callback.
```rust
    /// Fires all registered callbacks.
    pub fn run_all(&mut self, t:T) {
        self.clear_unused_callbacks();
        self.callback_list.iter_mut().for_each(move |(_,callback)| callback(t));
    }
```

Is there any suggestion you would like to make about this API?